daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Anf.scala

```diff
@@ -178,7 +176,7 @@ private[lf] object Anf {
 
   private def convertLoc(x: source.SELoc): target.SELoc = {
     x match {
-      case source.SELocS(x) => target.SELocS(x)
+      case source.SELocAbsoluteS(_) => sys.error("Anf.convertLoc/SELocAbsoluteS, unreachable code")
```

Comment:
should we use different types for the locations in different phases to avoid this?

Reply:
No, this is just an impossible code path. It was impossible before this change.
(_impossible_ in the sense than an earlier code branch has already dealt with this variant)

I improved the `Anf` code here (#11901) to avoid this impossible branch.